### ISSUE TITLE
Using a route-domain in route gateway address gives a 'bad declaration' error

### ISSUE THREAD

**Issue by erjac77** [F5Networks/f5-declarative-onboarding#140]
### Environment
 * Declarative Onboarding Version: 1.12.0
 * BIG-IP Version: BIG-IP 13.1.3.2 Build 0.0.4 Point Release 2 

### Summary
Trying to specify a route-domain for a route's gateway address gives a `bad declaration` error. Route-domain seems to be supported for the route's network, but not for the route's gateway address.

### Steps To Reproduce
1. Submit the following declaration:
```json
{
    "schemaVersion": "1.0.0",
    "class": "Device",
    "async": true,
    "label": "Base configuration for BIG-IP",
    "Common": {
        "class": "Tenant",
        "hostname": "bigip1.localhost",
        "internal-vlan": {
            "class": "VLAN",
            "tag": 4093,
            "mtu": 1500,
            "interfaces": [
                {
                    "name": "1.2",
                    "tagged": false
                }
            ]
        },
        "route-domain-10": {
            "class": "RouteDomain",
            "id": 10,
            "vlans": ["internal-vlan"]
        },
        "internal-self": {
            "class": "SelfIp",
            "address": "192.168.150.11%10/24",
            "vlan": "internal-vlan",
            "allowService": "none",
            "trafficGroup": "traffic-group-local-only"
        },
        "internal-route": {
            "class": "Route",
            "gw": "192.168.150.1%10",
            "network": "10.0.0.0%10/8"
        }
    }
}
```

2. Observe the following error response:
```json
{
    "id": "89f5e9b4-697b-4e3b-aa19-7647a77a6816",
    "selfLink": "https://localhost/mgmt/shared/declarative-onboarding/task/89f5e9b4-697b-4e3b-aa19-7647a77a6816",
    "code": 400,
    "status": "ERROR",
    "message": "bad declaration",
    "errors": [
        {
            "keyword": "format",
            "dataPath": ".declaration.Common['internal-route'].gw",
            "schemaPath": "network.schema.json#/allOf/3/then/properties/gw/anyOf/0/format",
            "params": {
                "format": "ipv4"
            },
            "message": "should match format \"ipv4\""
        },
        {
            "keyword": "format",
            "dataPath": ".declaration.Common['internal-route'].gw",
            "schemaPath": "network.schema.json#/allOf/3/then/properties/gw/anyOf/1/format",
            "params": {
                "format": "ipv6"
            },
            "message": "should match format \"ipv6\""
        },
        {
            "keyword": "anyOf",
            "dataPath": ".declaration.Common['internal-route'].gw",
            "schemaPath": "network.schema.json#/allOf/3/then/properties/gw/anyOf",
            "params": {},
            "message": "should match some schema in anyOf"
        }
    ],
    "result": {
        "class": "Result",
        "code": 400,
        "status": "ERROR",
        "message": "bad declaration",
        "errors": [
            {
                "keyword": "format",
                "dataPath": ".declaration.Common['internal-route'].gw",
                "schemaPath": "network.schema.json#/allOf/3/then/properties/gw/anyOf/0/format",
                "params": {
                    "format": "ipv4"
                },
                "message": "should match format \"ipv4\""
            },
            {
                "keyword": "format",
                "dataPath": ".declaration.Common['internal-route'].gw",
                "schemaPath": "network.schema.json#/allOf/3/then/properties/gw/anyOf/1/format",
                "params": {
                    "format": "ipv6"
                },
                "message": "should match format \"ipv6\""
            },
            {
                "keyword": "anyOf",
                "dataPath": ".declaration.Common['internal-route'].gw",
                "schemaPath": "network.schema.json#/allOf/3/then/properties/gw/anyOf",
                "params": {},
                "message": "should match some schema in anyOf"
            }
        ]
    },
    "declaration": {
        "schemaVersion": "1.0.0",
        "class": "Device",
        "async": true,
        "label": "Base configuration for BIG-IP",
        "Common": {
            "class": "Tenant",
            "hostname": "bigip1.localhost",
            "internal-vlan": {
                "class": "VLAN",
                "tag": 4093,
                "mtu": 1500,
                "interfaces": [
                    {
                        "name": "1.2",
                        "tagged": false
                    }
                ],
                "cmpHash": "default"
            },
            "route-domain-10": {
                "class": "RouteDomain",
                "id": 10,
                "vlans": [
                    "internal-vlan"
                ],
                "connectionLimit": 0,
                "strict": true
            },
            "internal-self": {
                "class": "SelfIp",
                "address": "192.168.150.11%10/24",
                "vlan": "internal-vlan",
                "allowService": "none",
                "trafficGroup": "traffic-group-local-only"
            },
            "internal-route": {
                "class": "Route",
                "gw": "192.168.150.1%10",
                "network": "10.0.0.0%10/8"
            }
        }
    }
}
```

### Expected Behavior
We should be able to specify a route-domain in the route's gateway address.

TMSH equivalent command:
```
tmsh create net route internal-route gw 192.168.150.1%10 network 10.0.0.0%10/8
```

### Actual Behavior
We receive a `bad declaration` error.
`"message": "should match format \"ipv4\""`

**Comment by dstokesf5** [comment]
Thank you for reporting this. I have reproduced the issue and added it to our internal product backlog as AUTOTOOL-1798.

**Comment by emmanuelgirard** [comment]
Also waiting for this fix ;) Thanks @erjac77 for reporting it !

**Comment by dstokesf5** [comment]
This issue was resolved as part of the 1.17.0 release.
